homeassistant/components/media_player/plex.py

```diff
@@ -272,93 +646,309 @@ def _active_media_plexapi_type(self):
-        self.device.skipNext(self._active_media_plexapi_type)
+        if self.device:
+            self._local_client_control_fix()
+            self.device.skipNext(self._active_media_plexapi_type)
 
     def media_previous_track(self):
         """Send previous track command."""
-        self.device.skipPrevious(self._active_media_plexapi_type)
+        if self.device:
+            self._local_client_control_fix()
+            self.device.skipPrevious(self._active_media_plexapi_type)
+
+    # pylint: disable=W0613
+    def play_media(self, media_type, media_id, **kwargs):
+        """Play a piece of media."""
+        if self.device:
+            src = json.loads(media_id)
+
+            media = None
+            if media_type == 'MUSIC':
+                media = self.device.server.library.section(
+                    src['library_name']).get(src['artist_name']).album(
+                        src['album_name']).get(src['track_name'])
+            elif media_type == 'EPISODE':
+                media = self._get_episode(
+                    src['library_name'], src['show_name'],
+                    src['season_number'], src['episode_number'])
+            elif media_type == 'PLAYLIST':
+                media = self.device.server.playlist(src['playlist_name'])
+            elif media_type == 'VIDEO':
+                media = self.device.server.library.section(
+                    src['library_name']).get(src['video_name'])
+
+            if media:
+                self._client_play_media(media, shuffle=src['shuffle'])
+
+    def _get_episode(self, library_name, show_name, season_number,
+                     episode_number):
+        """Find TV episode and return a Plex media object."""
+        target_season = None
+        target_episode = None
+
+        seasons = self.device.server.library.section(library_name).get(
+            show_name).seasons()
+        for season in seasons:
+            if int(season.seasonNumber) == int(season_number):
+                target_season = season
+                break
+
+        if target_season is None:
+            _LOGGER.error('Season not found: %s\\%s - S%sE%s', library_name,
+                          show_name,
+                          str(season_number).zfill(2),
+                          str(episode_number).zfill(2))
+        else:
+            for episode in target_season.episodes():
+                if int(episode.index) == int(episode_number):
+                    target_episode = episode
+                    break
+
+            if target_episode is None:
+                _LOGGER.error('Episode not found: %s\\%s - S%sE%s',
+                              library_name, show_name,
+                              str(season_number).zfill(2),
+                              str(episode_number).zfill(2))
+
+        return target_episode
+
+    def _client_play_media(self, media, **params):
+        """Instruct Plex client to play a piece of media."""
+        if self.device:
```

Comment:
`self.device` is not enough to know if it supports remote commands - needs to include `playback` in the `device.protocolCapabilities` (which still doesn't tell you if it supports volume controls.